hw/templates/snax_streamer_wrapper.sv.tpl

```diff
@@ -0,0 +1,159 @@
+// Copyright 2024 KU Leuven.
+// Solderpad Hardware License, Version 0.51, see LICENSE for details.
+// SPDX-License-Identifier: SHL-0.51
+
+<%
+  num_input_ports = len(cfg["snax_streamer_cfg"]["data_reader_params"]["tcdm_ports_num"])
+  num_output_ports = len(cfg["snax_streamer_cfg"]["data_writer_params"]["tcdm_ports_num"])
+  num_tcdm_ports = num_input_ports + num_output_ports
+
+  # We make the assumption that all reader and writers
+  # Have the same data widths
+  stream_data_width = cfg["snax_streamer_cfg"]["data_reader_params"]["element_width"][0]
+%>
+//-----------------------------
+// Streamer wrapper
+//-----------------------------
+module ${cfg["tag_name"]}_streamer_wrapper #(
+  // Parameters related to TCDM
+  parameter int unsigned NarrowDataWidth   = ${cfg["tcdm_data_width"]},
+  parameter int unsigned TCDMDepth         = ${cfg["tcdm_depth"]},
+  parameter int unsigned TCDMReqPorts      = ${num_tcdm_ports},
+  parameter int unsigned NrBanks           = ${cfg["num_banks"]},
+  parameter int unsigned TCDMSize          = NrBanks * TCDMDepth * (NarrowDataWidth/8),
+  parameter int unsigned TCDMAddrWidth     = $clog2(TCDMSize),
+  // Parameters related to streamers
+  // Touch at your own risk!
+  parameter int unsigned NumInputPorts     = ${num_input_ports},
+  parameter int unsigned NumOutputPorts    = ${num_output_ports},
+  parameter int unsigned StreamerDataWidth = ${stream_data_width}
+)(
+  //-----------------------------
+  // Clocks and reset
+  //-----------------------------
+  input  logic clk_i,
+  input  logic rst_ni,
+
+  //-----------------------------
+  // Accelerator ports
+  //-----------------------------
+  // Input ports from accelerator to streamer
+  input  logic [ NumInputPorts-1:0][StreamerDataWidth-1:0] acc2stream_data_i,
+  input  logic [ NumInputPorts-1:0]                        acc2stream_valid_i,
+  output logic [ NumInputPorts-1:0]                        acc2stream_ready_o,
```

Comment:
Bug!! Same issue. What is the exact meaning of `NumInputPorts` and `NumOutputPorts`. They have double meaning now and will cause functional errors!